### PR TITLE
PR: Fix history browsing while debugging

### LINKED ISSUE #5202: In ipdb, the up and down keys have a strange behaviour
## Description

**What steps will reproduce the problem?**

1. debug
2. press up to access the last thing you typed
3. wonder why you ended up in the middle of the history log

## Version and main components

* Spyder Version: 3.2.4.dev0 474e59379
* Python Version: 3.6.2
* Qt Versions: 5.9.1, PyQt5 5.9 on Darwin

## Dependencies
```
pyflakes >=0.6.0 :  1.6.0 (OK)
pycodestyle >=2.3:  2.3.1 (OK)
pygments >=2.0   :  2.2.0 (OK)
pandas >=0.13.1  :  0.20.3 (OK)
numpy >=1.7      :  1.13.1 (OK)
sphinx >=0.6.6   :  1.6.3 (OK)
rope >=0.9.4     :  0.10.5 (OK)
jedi >=0.9.0     :  0.10.2 (OK)
psutil >=0.3     :  5.2.2 (OK)
nbconvert >=4.0  :  5.2.1 (OK)
sympy >=0.7.3    :  1.1.1 (OK)
cython >=0.21    :  0.26 (OK)
qtconsole >=4.2.0:  4.3.0 (OK)
IPython >=4.0    :  6.1.0 (OK)
pylint >=0.25    :  1.7.2 (OK)

```

### PR DESCRIPTION
**Pull request by impact27**
Fixes #5202 

------------------------------------------
Edit:
Add reference to the issue

### CODE REVIEW

**Comment by dalthviz** [comment]
Hi @impact27, thanks a lot for working in this one! :+1:

**Comment by dalthviz** [comment]
Also could you please move this changes to `3.x`?, you could do it with the steps at our [Contributing guide](https://github.com/spyder-ide/spyder/blob/master/CONTRIBUTING.md#changing-base-branch)

Thanks! :)

**Comment by ccordoba12** [comment]
@impact27, no need to move this against 3.x. I think you just forgot to change the PR's base when opening it, but I fixed that for you.

File: spyder/widgets/mixins.py [hunk 1]
```diff
@@ -813,3 +813,7 @@ def find_in_history(self, tocursor, start_idx, backward):
                     return entry[len(tocursor):], idx
             else:
                 return None, start_idx
+
+    def reset_search_pos(self):
+        """Reset the position from witch to search the history"""
```
Comment (ccordoba12) [inline spyder/widgets/mixins.py]: Please change `witch` to `which`

**Comment by ccordoba12** [comment]
@dalthviz, please review that this works for you locally. This problem happened after introducing several commands while debugging (e.g. `c`, then `n`, then `!zz=10`) and pressing the Up arrow to get the previous entry in history.

**Comment by dalthviz** [comment]
@ccordoba12 It works for me locally 👍 

debug

Thanks again @impact27 for working on this one!

**Comment by ccordoba12** [comment]
Great! Thanks @impact27 for your contribution!

**Pull request merged**